### PR TITLE
Run test suites nightly

### PR DESCRIPTION
**Pull request by CohenArthur**

### CODE REVIEW

**Comment by CohenArthur** [comment]
So the testsuite generation + running lasted more than 6 hours :D Let's avoid regenerating the testsuites each time as that only matters for the `rustc-dejagnu` pass and see how that improves things.

**Comment by CohenArthur** [comment]
Remaining:

- [ ] Run each testsuite in a matrix
- [ ] Generate one result file per testsuite
- [ ] Make the result file a valid json object

**Comment by CohenArthur** [comment]
> commit [b31c94a](https://github.com/Rust-GCC/testing/pull/16/commits/b31c94ac222b37d33ee496fba8818f4ee64cbb95) copies in all of the rustc test files. Is this necessary and if so why? It makes this PR impossible to review.

Sorry, this PR should really have been a draft. This was an attempt at *not* running past the 6h point for our CI by not generating testsuites, which takes a lot of time. Sadly, this has proven to be fruitless as we are still starving the runner and getting the jobs canceled.

We'll need to use a custom runner somewhere, and I will revert that commit at that point and instead just run through the whole generation pass nightly which makes more sense.

**Comment by CohenArthur** [comment]
- [ ] `gccrs-success-no-core` has no results line?
- [ ] Cannot find the gccrs commit? Is it not `cat /GCCRS_COMMIT`?

**Comment by CohenArthur** [comment]
The rustc-dejagnu if guard does not work, but we will remove it anyway at some point

**Comment by CohenArthur** [comment]
We should maybe try adding a default timeout to the various testsuites. Right now on the runner, a single `rust1` job has been running for 7m30s

**Comment by CohenArthur** [comment]
This did not resolve the issue. Maybe we are running out of RAM or something? We should try adding some swap to the runner

**Comment by CohenArthur** [comment]
Or running it on a different runner with more RAM and see if that makes a difference.

**Comment by dkm** [comment]
> I'll unsubscribe from this PR for now. Feel free to ping me, once this is ready for review upside_down_face

Same here :)

File: .github/workflows/nightly_run.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        testsuite: [rustc-dejagnu, gccrs-parsing, gccrs-rustc-success, gccrs-rustc-success-no-std, gccrs-rustc-success-no-core, blake3, libcore-1.49]
+    steps:
+      - name: Fetch dependencies
+        run: |
+          # FIXME: Switch to `cargo install ftf` on next release, we need the
+          # `result-fmt` feature
+          cargo install --git https://github.com/cohenarthur/ftf
+
+      - uses: actions/checkout@v2
+      - name: Generate testsuites (nightly)
```
Comment (flip1995) [inline .github/workflows/nightly_run.yml:22]:
```suggestion
      - uses: actions/checkout@v3

      - name: Generate testsuites (nightly)
```

There's v3 of the checkout action IIRC. And lets insert newlines between steps.

File: .github/workflows/nightly_run.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          git submodule update --init
+          target/release/testsuite-adaptor --gccrs rust1 --rustc rustc \
+              --output-dir output-dir-${{ matrix.testsuite }} \
+              --yaml ${{ matrix.testsuite }}.yaml \
+              --rust-path rust --gccrs-path local_gccrs \
+              --passes ${{ matrix.testsuite }}
+
+      - name: Run testsuite
+        run: |
+          echo "{ \"name\": \"${{matrix.testsuite}}\", \"commit\": \"$(cat /GCCRS_BUILD)\", \"date\": \"$(date -I)\", \"results\": " >> ${{ matrix.testsuite }}.json
```
Comment (flip1995) [inline .github/workflows/nightly_run.yml:35]: Writing this to the file here is a bit ugly, I wouldn't block the PR on it though.

**Review by flip1995 — APPROVED** [review]
Only one small comment.  I like the "Files changed (1)" part and that all of it runs in 10 min.

**Pull request merged**